mayastor/tests/lock_lba_range.rs

```diff
@@ -0,0 +1,269 @@
+#[macro_use]
+extern crate log;
+
+pub mod common;
+
+use crossbeam::channel::unbounded;
+use mayastor::{
+    bdev::{nexus_create, nexus_lookup},
+    core::{
+        Bdev,
+        DmaBuf,
+        IoChannel,
+        MayastorCliArgs,
+        MayastorEnvironment,
+        RangeContext,
+        Reactor,
+        Reactors,
+    },
+};
+use std::sync::{Arc, Mutex};
+
+const NEXUS_NAME: &str = "lba_range_nexus";
+const NEXUS_SIZE: u64 = 10 * 1024 * 1024;
+const NUM_NEXUS_CHILDREN: u64 = 2;
+
+fn test_ini() {
+    test_init!();
+    for i in 0 .. NUM_NEXUS_CHILDREN {
+        common::delete_file(&[get_disk(i)]);
+        common::truncate_file_bytes(&get_disk(i), NEXUS_SIZE);
+    }
+
+    Reactor::block_on(async {
+        create_nexus().await;
+    });
+}
+
+fn test_fini() {
+    for i in 0 .. NUM_NEXUS_CHILDREN {
+        common::delete_file(&[get_disk(i)]);
+    }
+
+    Reactor::block_on(async {
+        let nexus = nexus_lookup(NEXUS_NAME).unwrap();
+        nexus.destroy().await.unwrap();
+    });
+}
+
+fn get_disk(number: u64) -> String {
+    format!("/tmp/disk{}.img", number)
+}
+
+fn get_dev(number: u64) -> String {
+    format!("aio://{}?blk_size=512", get_disk(number))
+}
+
+async fn create_nexus() {
+    let mut ch = Vec::new();
+    for i in 0 .. NUM_NEXUS_CHILDREN {
+        ch.push(get_dev(i));
+    }
+
+    nexus_create(NEXUS_NAME, NEXUS_SIZE, None, &ch)
+        .await
+        .unwrap();
+}
+
+fn get_shareable_ctx(offset: u64, len: u64) -> Arc<Mutex<RangeContext>> {
+    Arc::new(Mutex::new(RangeContext::new(offset, len)))
+}
+
+fn get_shareable_nexus_channel() -> Arc<Mutex<IoChannel>> {
+    let nexus = Bdev::open_by_name(NEXUS_NAME, true).unwrap();
+    Arc::new(Mutex::new(nexus.get_channel().unwrap()))
+}
+
+async fn lock_range(ctx: &mut RangeContext, ch: &IoChannel) {
+    let mut nexus = Bdev::open_by_name(NEXUS_NAME, true).unwrap();
+    let _ = nexus.lock_lba_range(ctx, ch).await;
+}
+
+async fn unlock_range(ctx: &mut RangeContext, ch: &IoChannel) {
+    let mut nexus = Bdev::open_by_name(NEXUS_NAME, true).unwrap();
+    let _ = nexus.unlock_lba_range(ctx, ch).await;
+}
+
+#[test]
+// Test acquiring and releasing a lock.
+fn lock_unlock() {
+    test_ini();
+    Reactor::block_on(async {
+        let mut nexus = Bdev::open_by_name(NEXUS_NAME, true).unwrap();
+        let mut ctx = RangeContext::new(1, 5);
+        let ch = nexus.get_channel().unwrap();
+        let _ = nexus.lock_lba_range(&mut ctx, &ch).await;
```

Comment:
You get back here (and ingore) a result<(), Error> -- what should it be? Ok() or Err() we should test for whatever is needed here?

Reply:
Done